rfcs/convergence/root-as-slot.md

```diff
@@ -0,0 +1,248 @@
+2. Support for Custom Elements on _slots_ but not for _root_
+3. Right now `Typings` for `root` aren't available as there's no way to split `root` from internal state.
+4. There's no possible way of passing refs to slots but `root` does
+5. [Primary slot]
+
+Major differences between `root` 🌿 and _slots_ 🎰:
+
+1. References 🌿
+2. Custom Elements as base element 🎰
+3. `as` prop 🌿
+4. Well defined Types 🎰
+
+## Detailed Design or Proposal
+
+For the User of the component perspective there will be no changes in the signature.
+
+### Converge `root` to a _slot_
+
+This RFC proposes to treat `root` as a _slot_.
+
+Treating `root` as a slot means having to declare `root` _Typings_ together with other _slots_. This will impact in a components state signature, where `root` will have to be declared.
+
+> ⚠️ That doesn't mean `root` will be available in component's properties interface
+
+What we have right now:
+
+```ts
+export type Slots = {
+  slot: ObjectShorthandProps<SlotProps>;
+};
+export interface Props extends ComponentProps<Slots> & React.HTMLAttributes<HTMLElement> {/* ... */}
+export interface State extends ComponentState<Slots> {/* ... */}
+
+// use*State hook
+function useState({ slot, ...props }: Props): State {
+  // ...
+  return {
+    ...props,
+    slot: resolveShorthand(slot),
+  };
+}
+```
+
+With the modifications:
+
+```ts
+export type Slots = {
+  root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+  slot: ObjectShorthandProps<SlotProps>;
+};
+export interface Props extends ComponentProps<Slots> {
+  /* ... */
+}
+export interface State extends ComponentState<Slots> {
+  /* ... */
+}
+
+// use*State hook
+function useState({slot, ...props}: Props): State {
+  // ...
+  return {
+    slot: resolveShorthand(slot)
+    root: resolveShorthand(props),
+  };
+}
+```
+
+### Make _slots_ support `as` property (but with restrictions)
+
+Cancels [Do not support as prop for slots that render native DOM elements] as it already conflicts with [1st rule of ARIA opt-out mechanism]. This would add `as` property for `ObjectShorthandProps` signature for all _slots_.
+
+To ensure `as` property is used for opt-out mechanism only native elements will be supported. _Typings_ must be used to ensure which native elements are supported.
+
+`ObjectShorthandProps` should be a model that _shorthand_ properties should extend from, and to ensure `as` property is only provided in cases that opt-out mechanism is required the `never` type should be used to disable `as` property on default cases.
+
+```ts
+type ObjectShorthandProps<Props = {}, Ref = unknown, As extends keyof JSX.IntrinsicElements = never> = Props &
+  React.RefAttributes<Ref> & {
+    as?: As;
+    children?: Props['children'] | ShorthandRenderFunction<Props>;
+  };
+```
+
+The case for `useARIAButton` is a good example of using this signature:
+
+```ts
+export type ARIAButtonProps =
```

Comment:
```suggestion
export type ARIAButtonShorthandProps =
```
As everything below is `ObjectShorthandProps`, IMO it should contain `Shorthand` in the name.